zilliqa/src/api/types/zil.rs

```diff
@@ -33,12 +33,87 @@ pub struct TxBlock {
 }
 
 impl TxBlock {
-    pub fn new(block: &Block, proposer: Address) -> Self {
-        // TODO(#79): Lots of these fields are empty/zero and shouldn't be.
+    pub fn new(block: &Block) -> Self {
         let mut scalar = [0; 32];
         scalar[31] = 1;
         TxBlock {
             header: TxBlockHeader {
+                version: 1,                                    // To match ZQ1
+                gas_limit: ScillaGas::from(block.gas_limit()), // In Scilla
+                gas_used: ScillaGas::from(block.gas_used()),   // In Scilla
+                rewards: 0,
+                txn_fees: 0,
+                prev_block_hash: block.parent_hash().into(),
+                block_num: block.number(),
+                timestamp: block
+                    .timestamp()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap_or_default()
+                    .as_micros(),
+                mb_info_hash: B256::ZERO, // Appears obsolete in ZQ2
```

Comment:
A bit nitpicky, but I'd prefer if the language of these comments was more assertive :) The current wording leads to some doubt when reading the code as to whether there is more work to do here to set this values differently.

I agree with all your observations and would be fine with just removing "appears" from them all :)

Reply:
Done